src/clic.adoc

```diff
@@ -912,104 +960,101 @@ through the trap handling process.
+ 26:24  (reserved)
+ 23:16  mpil[7:0]     Previous interrupt level
  15:12  (reserved)
- 11:0  Exccode[11:0] Exception/interrupt code
+ 11:0   exccode[11:0] Exception/interrupt code
 ----
 
-The `mcause.mpp` and `mcause.mpie` fields mirror the `mstatus.mpp` and
+The `mcause.mpp` and `mcause.mpie` fields typically mirror the `mstatus.mpp` and
 `mstatus.mpie` fields, and are aliased into `mcause` to reduce context
-save/restore code.
-
-Note: In a straightforward implementation, reading or writing mstatus fields mpp/mpie in mcause is equivalent to reading or writing the homonymous field in mstatus.
+save/restore code. For backwards compatibility in implementations supporting both CLINT and CLIC modes, when
+switching to CLINT mode the new CLIC {cause} state field
+({pil}) is zeroed.  The other new CLIC {cause} fields,
+{pp} and {pie}, appear as zero in the {cause} CSR but the corresponding
+state bits in the `mstatus` register are not cleared.
 
 If the hart is currently running at some privilege mode (`pp`) at some
 interrupt level (`pil`) and an enabled interrupt becomes pending at
-any interrupt level in a higher privilege mode or if an interrupt at a
+any interrupt level in a higher privilege mode or if an interrupt assigned to a
 higher interrupt level in the current privilege mode becomes pending
 and interrupts are globally enabled in this privilege mode, then
 execution is immediately transferred to a handler running with the new
 interrupt's privilege mode (`**__x__**`) and interrupt level (`il`).
 
 As stated in the RISC-V privilege specification, 
 when a trap is taken from privilege mode y into privilege mode x, 
-xPIE is set to the value of x IE; xIE is set to 0; and
+xPIE is set to the value of xIE; xIE is set to 0; and
 xPP is set to y.  xepc is written with the virtual address of the instruction
 that was interrupted or that encountered the exception.  
 Additionally in CLIC mode, interrupt level (`xpil`) is set to xintstatus.xil and
 xcause.exccode is written with a code indicating the event (the id of the
 interrupt or exception code) that caused the trap.
 
-For backwards compatibility in systems supporting both CLINT and CLIC modes, when
-switching to CLINT mode the new CLIC {cause} state field
-({pil}) is zeroed.  The other new CLIC {cause} fields,
-{pp} and {pie}, appear as zero in the {cause} CSR but the corresponding
-state bits in the `mstatus` register are not cleared.
-
-Note: For now all privilege modes must run in either CLIC mode or all privilege modes must run in non-CLIC mode so switching to CLINT mode from CLIC mode causes {pil} in all privilege modes to be zeroed.
+Note: For now all privilege modes must run in either CLIC mode or all 
+privilege modes must run in non-CLIC mode so switching to CLINT mode 
+from CLIC mode causes {pil} in all privilege modes to be zeroed.
 
-when not in CLIC mode, {cause} has the CLINT mode format.
+When not in CLIC mode, {cause} has the CLINT mode format.
```

Comment:
should be xcause?

Reply:
No, the ASCIIDOC converts {cause} to **x**cause automatically. BTW, this happens everywhere and for other CSR names too.